File: src/main/kotlin/br/com/fiap/techfood/adapters/outbound/repository/entities/OrderItemPk.java
package br.com.fiap.techfood.adapters.outbound.repository.entities;


import jakarta.persistence.Embeddable;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

import java.util.Objects;

@Embeddable
public class OrderItemPk {

    @ManyToOne
    @JoinColumn(name = "order_id")
    private OrderEntity orderEntity;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private ProductEntity product;

    public OrderEntity getOrderEntity() {
        return orderEntity;
    }

    public void setOrderEntity(OrderEntity orderEntity) {
        this.orderEntity = orderEntity;
    }

    public ProductEntity getProduct() {
        return product;
    }

    public void setProduct(ProductEntity product) {
        this.product = product;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        OrderItemPk that = (OrderItemPk) o;
        return Objects.equals(orderEntity, that.orderEntity) && Objects.equals(product, that.product);
    }

    @Override
    public int hashCode() {
        return Objects.hash(orderEntity, product);
    }
}

File: src/main/kotlin/br/com/fiap/techfood/adapters/outbound/repository/entities/OrderItemPk.kt
package br.com.fiap.techfood.adapters.outbound.repository.entities

import jakarta.persistence.Embeddable
import jakarta.persistence.ForeignKey
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.io.Serializable
import java.util.*

@Embeddable
class OrderItemPk(
    @ManyToOne
    @JoinColumn(name = "order_id",  foreignKey = ForeignKey(name = "fk_order_item_order"))
    var order: OrderEntity? = null,

    @ManyToOne
    @JoinColumn(name = "product_id",  foreignKey = ForeignKey(name = "fk_order_item_product"))
    var product: ProductEntity? = null
) : Serializable {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as OrderItemPk
        return order == that.order && product == that.product
    }

    override fun hashCode(): Int {
        return Objects.hash(order, product)
    }
}
